solr/core/src/java/org/apache/solr/cloud/api/collections/OverseerCollectionMessageHandler.java

```diff
@@ -554,12 +554,11 @@ private void modifyCollection(ClusterState clusterState, ZkNodeProps message, @S
     final String collectionName = message.getStr(ZkStateReader.COLLECTION_PROP);
     //the rest of the processing is based on writing cluster state properties
     //remove the property here to avoid any errors down the pipeline due to this property appearing
-    String configName = (String) message.getProperties().remove(CollectionAdminParams.COLL_CONF);
+    String configName = (String) message.getProperties().get(CollectionAdminParams.COLL_CONF);
```

Comment:
just observing that this innocent looking change seems important to this PR.  Previously this data had disappeared from the state.